radix-engine-interface/src/api/field_api.rs

```diff
@@ -52,7 +56,9 @@ pub trait SystemFieldApi<E: Debug> {
         self.field_write(handle, buf)
     }
 
+    // Lock a field such that it is immutable
```

Comment:
You cannot get a read/write handle on an immutable field. And locking a field requires the handle to be a read/write handle, otherwise an error will occur.